File: frontend/api/photo.ts
import axios from 'axios';

interface UnsplashResponse {
    results: {
        urls: {
            raw: string;
        };
    }[];
}

async function imageGenerator(location: string): Promise<string | null> {
    // Your Unsplash API key
    const apiKey = '8lUqDkIz_wzBmP1humc8zMF2U76DM_P8tiJsGcAmWOE';
    // Text input for the image search
    const query = location;
    // API endpoint for searching photos
    const apiUrl = `https://api.unsplash.com/search/photos?query=${query}&client_id=${apiKey}`;

    try {
        // Send a GET request to the Unsplash API
        const response = await axios.get<UnsplashResponse>(apiUrl);
        // Get the URL of the first image from the search results
        const data = response.data;
        if (data.results && data.results.length > 0) {
            const imageUrl: string = data.results[0].urls.raw;
            return imageUrl;
        } else {
            console.error('No results found for the given location.');
            return null;
        }
    } catch (error: any) {
        console.error(`Error: ${(error as Error).message}`);
        return null;
    }
}

async function getImage(city: string) {
    const imageUrl = await imageGenerator(city);
    if (imageUrl) {
        console.log('Image URL:', imageUrl);
        // You can use imageUrl directly in your application or return it as needed
    } else {
        console.log('Image not found.');
    }
}

// Example usage
const city = 'Brussels';
getImage(city);

File: photo.py
import requests
import os


def photo_generator(location):
    # Define your API key
    api_key = os.environ.get("GOOGLE_API_KEY")
    # Step 1: Perform a Place Search
    search_endpoint = "https://maps.googleapis.com/maps/api/place/textsearch/json"
    search_params = {
        'query': f"{location}",
        'key': api_key
    }

    response = requests.get(search_endpoint, params=search_params)
    if response.status_code == 200:
        data = response.json()
        # Check if there are results
        if 'results' in data and len(data['results']) > 0:
            # Extract the first result's photo_reference (you can loop through results if needed)
            photo_reference = data['results'][0].get(
                'photos', [])[0].get('photo_reference', None)
            if photo_reference:
                # Step 2: Use the photo_reference to fetch the image
                photo_endpoint = "https://maps.googleapis.com/maps/api/place/photo"
                photo_params = {
                    'maxwidth': '400',
                    'photoreference': photo_reference,
                    'key': api_key
                }
                photo_response = requests.get(
                    photo_endpoint, params=photo_params)
                if photo_response.status_code == 200:
                    # Step 3: Save the image to a file
                    with open('photo.jpg', 'wb') as file:
                        file.write(photo_response.content)
                    print("Photo downloaded successfully.")
                else:
                    print(f"Error downloading photo: {
                          photo_response.status_code} - {photo_response.text}")
            else:
                print("No photo_reference found in the search results.")
        else:
            print("No results found for the search query.")
    else:
        print(f"Error searching for places: {
              response.status_code} - {response.text}")


photo_generator("Brussels")
